I've gotten this twice now but I didn't change anything there. Very confused.

```
19:58:26 
lib/libswiftSema.a(TypeCheckDecl.cpp.o):/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swift/lib/Sema/TypeCheckDecl.cpp:function (anonymous namespace)::DeclChecker::visitClassDecl(swift::ClassDecl*): error: undefined reference to 'swift::SerializedASTFile::getLanguageVersionBuiltWith() const'
19:58:26 
lib/libswiftSema.a(TypeCheckDecl.cpp.o):/home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/swift/lib/Sema/TypeCheckDecl.cpp:function (anonymous namespace)::DeclChecker::visitClassDecl(swift::ClassDecl*): error: undefined reference to 'swift::SerializedASTFile::getLanguageVersionBuiltWith() const'
```

@swift-ci Please clean test Linux

--- a/lib/ClangImporter/ClangImporter.cpp
+++ b/lib/ClangImporter/ClangImporter.cpp
@@ -2435,6 +2435,23 @@ static bool isVisibleClangEntry(clang::ASTContext &ctx,
 TypeDecl *
 ClangModuleUnit::lookupNestedType(Identifier name,
                                   const NominalTypeDecl *baseType) const {
+  // Special case for error code enums: try looking directly into the struct
+  // first. But only if it looks like a synthesized error wrapped struct.
+  if (name == getASTContext().Id_Code && !baseType->hasClangNode() &&
+      isa<StructDecl>(baseType) && !baseType->hasLazyMembers() &&
+      baseType->isChildContextOf(this)) {
+    auto *mutableBase = const_cast<NominalTypeDecl *>(baseType);
+    auto codeEnum = mutableBase->lookupDirect(name,/*ignoreNewExtensions*/true);
+    // Double-check that we actually have a good result. It's possible what we
+    // found is /not/ a synthesized error struct, but just something that looks
+    // like it. But if we still found a good result we should return that.
+    if (codeEnum.size() == 1 && isa<TypeDecl>(codeEnum.front()))
+      return cast<TypeDecl>(codeEnum.front());
+    if (codeEnum.size() > 1)
+      return nullptr;
+    // Otherwise, fall back and try via lookup table.
+  }
+
   auto lookupTable = owner.findLookupTable(clangModule);
   if (!lookupTable)
     return nullptr;
@@ -2491,9 +2508,13 @@ ClangModuleUnit::lookupNestedType(Identifier name,
     });
   }
 
-  if (results.size() == 1)
-    return results.front();
-  return nullptr;
+  if (results.size() != 1) {
+    // It's possible that two types were import-as-member'd onto the same base
+    // type with the same name. In this case, fall back to regular lookup.
+    return nullptr;
+  }
+
+  return results.front();
 }
 
 void ClangImporter::loadExtensions(NominalTypeDecl *nominal,

--- a/test/ClangImporter/Inputs/custom-modules/ImportAsMember.h
+++ b/test/ClangImporter/Inputs/custom-modules/ImportAsMember.h
@@ -57,4 +57,12 @@ extern void IAMStruct1SelfComesLast(double x, struct IAMStruct1 s)
 extern void IAMStruct1SelfComesThird(int a, float b, struct IAMStruct1 s, double x)
     __attribute__((swift_name("Struct1.selfComesThird(a:b:self:x:)")));
 
+
+struct IAMMultipleNested {
+  int value;
+};
+
+typedef int MNInnerInt __attribute__((swift_name("IAMMultipleNested.Inner")));
+typedef float MNInnerFloat __attribute__((swift_name("IAMMultipleNested.Inner")));
+
 #endif // IMPORT_AS_MEMBER_H

--- a/test/ClangImporter/Inputs/frameworks/ImportAsMemberSubmodules.framework/Headers/Actual.h
+++ b/test/ClangImporter/Inputs/frameworks/ImportAsMemberSubmodules.framework/Headers/Actual.h
@@ -1,3 +1,3 @@
-struct IAMOuter { int x; };
+struct IAMSOuter { int x; };
 
-struct IAMInner { int y; };
+struct IAMSInner { int y; };

--- a/test/ClangImporter/Inputs/frameworks/ImportAsMemberSubmodules.framework/Headers/Fwd.h
+++ b/test/ClangImporter/Inputs/frameworks/ImportAsMemberSubmodules.framework/Headers/Fwd.h
@@ -1,3 +1,3 @@
 // The order of these forward-declarations affects whether there was a bug.
-struct IAMOuter;
-struct IAMInner;
+struct IAMSOuter;
+struct IAMSInner;

--- a/test/ClangImporter/Inputs/frameworks/ImportAsMemberSubmodules.framework/Headers/ImportAsMemberSubmodules.apinotes
+++ b/test/ClangImporter/Inputs/frameworks/ImportAsMemberSubmodules.framework/Headers/ImportAsMemberSubmodules.apinotes
@@ -1,4 +1,4 @@
 Name: ImportAsMemberSubmodules
 Tags:
-- Name: IAMInner
-  SwiftName: IAMOuter.Inner
+- Name: IAMSInner
+  SwiftName: IAMSOuter.Inner

--- a/test/ClangImporter/import-as-member.swift
+++ b/test/ClangImporter/import-as-member.swift
@@ -1,5 +1,7 @@
-// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -F %S/Inputs/frameworks %s -verify
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -F %S/Inputs/frameworks -I %S/Inputs/custom-modules %s -verify
 
+import ImportAsMember
 import ImportAsMemberSubmodules
 
-let _: IAMOuter.Inner?
+let _: IAMSOuter.Inner?
+let _: IAMMultipleNested.Inner? // expected-error {{ambiguous type name 'Inner' in 'IAMMultipleNested'}}

--- a/test/Serialization/Inputs/xref-nested-clang-type/NestedClangTypes.h
+++ b/test/Serialization/Inputs/xref-nested-clang-type/NestedClangTypes.h
@@ -5,3 +5,14 @@ struct Outer {
 struct __attribute__((swift_name("Outer.InterestingValue"))) Inner {
   int value;
 };
+
+#if __OBJC__
+
+@import Foundation;
+
+extern NSString * const ErrorCodeDomain;
+enum __attribute__((ns_error_domain(ErrorCodeDomain))) ErrorCodeEnum {
+  ErrorCodeEnumA
+};
+
+#endif // __OBJC__

--- a/test/Serialization/xref-nested-clang-type.swift
+++ b/test/Serialization/xref-nested-clang-type.swift
@@ -1,8 +1,16 @@
 // RUN: %empty-directory(%t)
-
 // RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -emit-module -o %t -I %S/Inputs/xref-nested-clang-type/ %s -module-name Lib
-
 // RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -I %t -I %S/Inputs/xref-nested-clang-type/ %s -DCLIENT -verify
+
+// RUN: %empty-directory(%t)
+// RUN: cp %S/Inputs/xref-nested-clang-type/NestedClangTypes.h %t
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -emit-module -o %t -import-objc-header %t/NestedClangTypes.h %s -module-name Lib -DNO_MODULE
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -I %t -import-objc-header %t/NestedClangTypes.h %s -DCLIENT -verify
+
+// RUN: %empty-directory(%t)
+// RUN: cp %S/Inputs/xref-nested-clang-type/NestedClangTypes.h %t
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -emit-module -o %t -import-objc-header %t/NestedClangTypes.h -pch-output-dir %t/PCHCache %s -module-name Lib -DNO_MODULE
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -I %t -pch-output-dir %t/PCHCache -import-objc-header %t/NestedClangTypes.h %s -DCLIENT -verify
 
 #if CLIENT
 
@@ -17,7 +25,9 @@ func test(x: MyCode) {}
 
 #else // CLIENT
 
+#if !NO_MODULE
 import NestedClangTypes
+#endif
 
 public typealias MyOuter = Outer
 public typealias MyInner = Outer.InterestingValue

--- a/test/Serialization/xref-nested-clang-type.swift
+++ b/test/Serialization/xref-nested-clang-type.swift
@@ -1,8 +1,8 @@
 // RUN: %empty-directory(%t)
 
-// RUN: %target-swift-frontend -emit-module -o %t -I %S/Inputs/xref-nested-clang-type/ %s -module-name Lib
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -emit-module -o %t -I %S/Inputs/xref-nested-clang-type/ %s -module-name Lib
 
-// RUN: %target-swift-frontend -typecheck -I %t -I %S/Inputs/xref-nested-clang-type/ %s -DCLIENT -verify
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -I %t -I %S/Inputs/xref-nested-clang-type/ %s -DCLIENT -verify
 
 #if CLIENT
 
@@ -10,7 +10,12 @@ import Lib
 
 func test(x: MyInner) {}
 
-#else
+#if _runtime(_ObjC)
+import Foundation
+func test(x: MyCode) {}
+#endif // ObjC
+
+#else // CLIENT
 
 import NestedClangTypes
 
@@ -21,4 +26,13 @@ extension MyOuter {
   public func use(inner: MyInner) {}
 }
 
-#endif
+#if _runtime(_ObjC)
+import Foundation
+
+public typealias MyCode = ErrorCodeEnum.Code
+extension ErrorCodeEnum {
+  public func whatever(code: MyCode) {}
+}
+#endif // ObjC
+
+#endif // CLIENT

